tests/integration/test_agentd/test_agentd_state/test_agentd_state.py

```diff
@@ -0,0 +1,365 @@
+def test_agentd_state(test_configuration, test_metadata, set_wazuh_configuration, remove_state_file, configure_local_internal_options, truncate_monitored_files):
+    '''
+    description: Check that the statistics file 'wazuh-agentd.state' is created automatically
+                 and verify that the content of its fields is correct.
+
+    wazuh_min_version: 4.2.0
+
+    tier: 0
+
+    parameters:
+        - test_configuration:
+            type: data
+            brief: Configuration used in the test.
+        - test_metadata:
+            type: data
+            brief: Configuration cases.
+        - set_wazuh_configuration:
+            type: fixture
+            brief: Configure a custom environment for testing.
+        - remove_state_file:
+            type: fixture
+            brief: Removes the wazuh-agentd.state file
+        - configure_local_internal_options:
+            type: fixture
+            brief: Set internal configuration for testing.
+        - truncate_monitored_files:
+            type: fixture
+            brief: Reset the 'ossec.log' file and start a new monitor.
+
+    assertions:
+        - Verify that the 'wazuh-agentd.state' statistics file has been created.
+        - Verify that the information stored in the 'wazuh-agentd.state' statistics file
+          is consistent with the connection status to the 'wazuh-remoted' daemon.
+
+    input_description: An external YAML file (wazuh_conf.yaml) includes configuration settings for the agent.
+                       Different test cases that are contained in an external YAML file (wazuh_state_tests.yaml)
+                       that includes the parameters and their expected responses.
+
+    expected_output:
+        - r'pending'
+        - r'connected'
+    '''
+
+    # Stop service
+    control_service('stop')
+
+    # Start RemotedSimulator if test case need it
+    remoted_server = start_remoted_server(test_metadata) 
+
+    # Add dummy key in order to communicate with RemotedSimulator
+    add_custom_key()
+
+    # Start service
+    control_service('start')
+
+    # Check fields for every expected output type
+    for expected_output in test_metadata['output']:
+        check_fields(expected_output, remoted_server)
+
+    if remoted_server:
+        remoted_server.clear()
+        remoted_server.shutdown()
+    
+    
+def parse_state_file():
+    """Parse state file
+
+    Returns:
+        state info
+    """
+    # Wait until state file is dumped
+    wait_state_update()
+    state = {}
+    with open(AGENTD_STATE) as state_file:
+        for line in state_file:
+            line = line.rstrip('\n')
+            # Remove empty lines or comments
+            if not line or line.startswith('#'):
+                continue
+            (key, value) = line.split('=', 1)
+            # Remove value's quotes
+            state[key] = value.strip("'")
+
+    return state
+    
```

Comment:
Missing empty space. Fix this in all the file.

Reply:
Fixed in https://github.com/wazuh/wazuh/pull/18353/commits/ecf23d03eafffc97b67a2d364324e916eb50d143